source/common/http/http2/codec_impl.cc

```diff
@@ -474,6 +475,13 @@ int ConnectionImpl::onFrameReceived(const nghttp2_frame* frame) {
     return 0;
   }
 
+  // Only for tests, issue a callback when the SETTINGS frame is received.
+  if (frame->hd.type == NGHTTP2_SETTINGS && frame->hd.flags == NGHTTP2_FLAG_NONE) {
```

Comment:
mild preference for a no-op function onSettings() commented that it's only used for test.
Then I think you can do using onSettings in TestClientConnectionImpl and TestServerConnectionImpl to latch it, and eventually it can be reused by folks who want to consume settings frames.

Reply:
SG; this is definitely more future proof. Done.